draft-ietf-dnsop-svcb-https.md

```diff
@@ -1443,25 +1443,22 @@ ACTION: create and include a reference to this registry.
 
 A registration MUST include the following fields:
 
-* Number: SvcParamKey wire format numeric identifier (range 0-65535)
-* Name: SvcParamKey presentation name
+* Number: wire format numeric identifier (range 0-65535)
+* Name: unique presentation name
 * Meaning: a short description
-* Pointer to specification text
+* Reference: pointer to specification text
 
-SvcParamKey entries to be added to this namespace
-have different policies ({{!RFC8126}}, Section 4)
-based on their range:
+Apart from the initial contents, the SvcParamKey name MUST NOT start with "key".
 
-| Number      | IANA Policy             |
-| ----------- | ----------------------  |
-| 0-255       | Standards Action        |
-| 256-32767   | Expert Review           |
-| 32768-65280 | First Come First Served |
-| 65280-65534 | Private Use             |
-| 65535       | Standards Action        |
+Entries in this registry are subject to a Specification Required registration
+policy ({{!RFC8126}}, Section 4.6).  The provided specification MUST specify
+the SvcParamValue's presentation format and MAY detail its intended meaning
+and use.  Additionally, the registry SHALL allow registrants to have a Number
+temporarily marked as "Reserved" (without an associated Name)
+on a First Come First Served basis.
```

Comment:
This temporary first-come-first-served seems awkward. I think requiring the IANA registration to define the presentation format is much simpler and consistent with IANA policy.

Reply:
Does that mean you're OK with having to specify the presentation format before IANA allocates you a number?  Remember that FCFS doesn't mean you get to choose your number (except at IANA's discretion).

Reply:
Yeah, I think it's quite reasonable to have to give information like that on the allocation.

Reply:
The apparent goal of this text is to be able to claim a code-point for experimental purposes before the presentation format is fixed.  RFC 7120 requires any early allocations to be from within the IETF, and for the implementation to be fully stable.  FCFS-to-reserve is an end-run around those requirements.  (And something that IANA specifically says [does not exist](https://www.iana.org/help/protocol-registration):  "There is an early allocation process available, but there is no reservation system.")  Is the intended audience here non-IETF and/or unstable usage?  If not, let's just stick with early allocation.

Note that there is still a Private Use range, which would work as a place to implement until the specification is stable enough for EA.

Reply:
I don't think the FCFS range should be meant for code points that are intended to be later reserved and standardized. Rather, I'd expect them to be used for non-private use for specs that aren't fully standardized, but have value for deploying and support. Stable specs and RFCs would use the more official space once they have final formats, etc.

An example is how Oblivious DoH is using a value in the FCFS range with a specific definition that is used for interoperating implementations on the public Internet; it's well-defined, but it may not be the same as some later RFC that could eventually take precedence.